.github/actions/setup-python/action.yml

```diff
@@ -0,0 +1,60 @@
+name: 'python setup'
+description: 'Configures python, pip, and configures pip cache dir as output.'
+
+inputs:
+  cache-files-hash:
+    description: 'A single hash for a set of files. Used for caching.'
+    required: false
+    default: ${{ hashFiles('requirements-*.txt', '!requirements-pre-commit.txt') }}
```

Comment:
This seems to work as `backend-lint` hashes `requirements-pre-commit` only:

![image](https://user-images.githubusercontent.com/79684/131408302-3f6bb78a-990e-4170-a4c8-b52d0d95ce65.png)

and the other workflows have:

![image](https://user-images.githubusercontent.com/79684/131408341-08139f53-b982-4a85-964a-7a7217933b7c.png)

Reply:
Nice stuff 👍